homeassistant/components/wemo/wemo_device.py

```diff
@@ -0,0 +1,89 @@
+"""Home Assistant wrapper for a pyWeMo device."""
+import logging
+
+from pywemo import PyWeMoException, WeMoDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_NAME,
+    CONF_PARAMS,
+    CONF_TYPE,
+    CONF_UNIQUE_ID,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, WEMO_SUBSCRIPTION_EVENT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceWrapper:
+    """Home Assistant wrapper for a pyWeMo device."""
+
+    def __init__(self, hass: HomeAssistant, wemo: WeMoDevice, device_id: str) -> None:
+        """Initialize DeviceWrapper."""
+        self.hass = hass
+        self.wemo = wemo
+        self.device_id = device_id
+        self.device_info = _device_info(wemo)
+        self.supports_long_press = wemo.supports_long_press()
```

Comment:
Is this call async safe?

Reply:
It is, yes. It's just calling [`issubclass(cls, LongPressMixin)`](https://github.com/pywemo/pywemo/blob/75b2eb868fac06c64290ff02c256979562fad2f7/pywemo/ouimeaux_device/__init__.py#L642)